### ISSUE TITLE
Build Error  unknown type name 'QStringRef'; did you mean 'CFStringRef'?

### ISSUE THREAD

**Issue by Kalanamith** [bitcoin/bitcoin#21901]
Getting the following error when building
```
/usr/local/Cellar/qt@5/5.15.2/lib/QtCore.framework/Headers/qlocale.h:1033:25: error: 
      unknown type name 'QStringRef'; did you mean 'CFStringRef'?
    ulong toULong(const QStringRef &s, bool *ok = nullptr) const;
                        ^~~~~~~~~~
                        CFStringRef
/usr/local/include/QtCore/qdebug.h:473:1: note: 'CFStringRef' declared here
Q_FORWARD_DECLARE_CF_TYPE(CFString);

```

**Comment by fanquake** [comment]
Do you have qt 5 and 6 installed via brew? If so, you will need to uninstall qt6 before configuring / compiling.

**Comment by Kalanamith** [comment]
Yep install qt5 as per installation guide

**Comment by fanquake** [comment]
Not much more we can do here without more info. Can you provide a config.log or similar? Which version of the source are you building? etc. Will reopen when we've got more info.

**Comment by gcsfred2** [comment]
I have the same error. Is there an alternative that is not downgrading qt?

**Comment by hebasto** [comment]
@gcsfred2 

> Not much more we can do here without more info. Can you provide a config.log or similar? Which version of the source are you building? etc. Will reopen when we've got more info.

**Comment by da2ce7** [comment]
Hello @hebasto and @fanquake 

I get the same qt compiler error:

```
Making all in src
  CXX      qt/libbitcoinqt_a-bantablemodel.o
In file included from qt/bantablemodel.cpp:14:
In file included from /usr/local/Cellar/qt@5/5.15.5_1/lib/QtCore.framework/Headers/QLocale:1:
/usr/local/Cellar/qt@5/5.15.5_1/lib/QtCore.framework/Headers/qlocale.h:1028:25: error: unknown type name 'QStringRef'; did you mean 'CFStringRef'?
    short toShort(const QStringRef &s, bool *ok = nullptr) const;
                        ^~~~~~~~~~
                        CFStringRef
/usr/local/include/QtCore/qdebug.h:531:1: note: 'CFStringRef' declared here
Q_FORWARD_DECLARE_CF_TYPE(CFString);
^
/usr/local/include/QtCore/qglobal.h:802:77: note: expanded from macro 'Q_FORWARD_DECLARE_CF_TYPE'
#  define Q_FORWARD_DECLARE_CF_TYPE(type) typedef const struct __ ## type * type ## Ref
                                                                            ^
<scratch space>:112:1: note: expanded from here
CFStringRef
^
```

```bash
% git clean -ffxd
% git rev-parse HEAD
e191fac4f3c37820f0618f72f0a8e8b524531ab8
% ./autogen.sh
% ./configure
% make
```
output: https://gist.github.com/da2ce7/c697259e204c961876ad7f7057e95b34

config.log: https://gist.github.com/da2ce7/ec7276de7d0284681cd73b0cdcb90f3b

brew list --version --formulae: https://gist.github.com/da2ce7/25d6473224a1d69a28cf24e44abb55b7
